pkg/scheduler/plugins/loadaware/load_aware.go

```diff
@@ -146,75 +145,52 @@ func (p *Plugin) Filter(ctx context.Context, state *framework.CycleState, pod *c
 		return nil
 	}
 
+	allocatable, err := p.estimator.EstimateNode(node)
+	if err != nil {
+		return nil
+	}
 	filterProfile := generateUsageThresholdsFilterProfile(node, p.args)
-	if len(filterProfile.ProdUsageThresholds) > 0 && extension.GetPodPriorityClassWithDefault(pod) == extension.PriorityProd {
-		status := p.filterProdUsage(node, nodeMetric, filterProfile.ProdUsageThresholds)
-		if !status.IsSuccess() {
-			return status
-		}
+	prodPod := len(filterProfile.ProdUsageThresholds) > 0 && extension.GetPodPriorityClassWithDefault(pod) == extension.PriorityProd
+
+	var nodeUsage *slov1alpha1.ResourceMap
+	var usageThresholds map[corev1.ResourceName]int64
+	if prodPod {
+		usageThresholds = filterProfile.ProdUsageThresholds
 	} else {
-		var usageThresholds map[corev1.ResourceName]int64
+		if nodeMetric.Status.NodeMetric != nil {
+			if filterProfile.AggregatedUsage != nil {
```

Comment:
Here the cyclomatic complexity is too high. Please revise the logic for readability.
e.g. The `nodeUsage` is exactly a pre-processing variable that only gets used in the function `GetEstimatedUsed`. Perhaps it can be moved into the function.

Reply:
ok, I will move the nodemetric check to after obtaining this resource.